apps/client/src/features/viewers/timer/Timer.tsx

```diff
@@ -61,29 +63,62 @@ export default function Timer(props) {
   const clock = formatTime(time.clock, formatOptions);
   const showOverlay = pres.text !== '' && pres.visible;
   const isPlaying = time.playback !== 'pause';
-  const normalisedTime = time.current === null ? null : Math.max(time.current, 0);
+  const isNegative =
+    (time.current ?? 0) < 0 && time.timerType !== TimerType.Clock && time.timerType !== TimerType.CountUp;
+
+  const showEndMessage = time.current < 0 && general.endMessage;
   const showProgress = time.playback !== 'stop';
+  const showFinished = time.finished && (time.timerType !== TimerType.Clock || showEndMessage);
+  const showClock = time.timerType !== TimerType.Clock;
   const baseClasses = `stage-timer ${isMirrored ? 'mirror' : ''}`;
 
+  const stageTimer = getTimerByType(time);
+
+  let display = '';
+
+  if (typeof stageTimer === 'string') {
+    display = stageTimer;
+  } else if (stageTimer === null || typeof stageTimer === 'undefined' || isNaN(stageTimer)) {
+    display = '-- : -- : --';
+  } else {
+    display = formatDisplay(millisToSeconds(stageTimer), true);
+  }
+
+  if (isNegative) {
+    display = `-${display}`;
+  }
+
+  const stageTimerCharacters = display.replace('/:/g', '').length;
+
   return (
-    <div className={time.finished ? `${baseClasses} stage-timer--finished` : baseClasses} data-testid='timer-view'>
+    <div className={showFinished ? `${baseClasses} stage-timer--finished` : baseClasses} data-testid='timer-view'>
       <NavigationMenu />
       <div className={showOverlay ? 'message-overlay message-overlay--active' : 'message-overlay'}>
         <div className='message'>{pres.text}</div>
       </div>
 
-      <div className='clock-container'>
+      <div
+        className='clock-container'
+        style={{
+          opacity: `${showClock ? 1 : 0}`,
+        }}
+      >
         <div className='label'>Time Now</div>
         <div className='clock'>{clock}</div>
       </div>
 
-      <div className='timer-container'>
-        {time.finished ? (
-          <div className='end-message'>
-            {!general.endMessage ? <TimerDisplay time={time.current} hideZeroHours /> : general.endMessage}
-          </div>
+      <div
+        className='timer-container'
+        style={{
+          fontSize: `${100 / stageTimerCharacters}vw`,
+        }}
+      >
+        {showEndMessage ? (
+          <div className='end-message'>{general.endMessage}</div>
         ) : (
-          <TimerDisplay time={normalisedTime} hideZeroHours className={isPlaying ? 'timer' : 'timer--paused'} />
+          <div className={`timer ${!isPlaying ? 'timer--paused' : ''} ${showFinished ? 'timer--finished' : ''}`}>
```

Comment:
I guess this is the way to go, few suggestions:
- lets make the calculation for fontsize in the body of the function, before the return
- same with the classname, lets do it up on the body